test/e2e/network/service.go

```diff
@@ -1682,10 +1682,32 @@ var _ = common.SIGDescribe("Services", func() {
 
 		service := t.BuildServiceSpec()
 		service.Spec.Type = v1.ServiceTypeNodePort
-
+		errAllocated := errors.New("provided port is already allocated")
+		noNodePortAvailable := errors.New("no available nodeport found")
 		ginkgo.By("creating service " + serviceName + " with type NodePort in namespace " + ns)
-		service, err := t.CreateService(service)
+		err := retry.OnError(retry.DefaultRetry, func(e error) bool {
+			if e == nil {
+				return false
+			}
+			if strings.Contains(e.Error(), errAllocated.Error()) ||
+				strings.Contains(e.Error(), noNodePortAvailable.Error()) {
+				return true
+			}
+			return false
+		}, func() error {
+			v, ok := e2eservice.GetUnusedStaticNodePort()
+			if !ok {
+				return noNodePortAvailable
+			}
+			service.Spec.Ports[0].NodePort = int32(v)
+			var err error
+			service, err = t.CreateService(service)
+			return err
+		})
 		framework.ExpectNoError(err, "failed to create service: %s in namespace: %s", serviceName, ns)
+		nodePort := service.Spec.Ports[0].NodePort
+		e2eservice.ReserveStaticNodePort(int(nodePort))
```

Comment:
this returns a boolean, you should check the result and log it , so we can troublshoot problems , per example, the service was created succesfuly but we failed to reserve the port
```suggestion
		ok := e2eservice.ReserveStaticNodePort(int(nodePort))
		if !ok {
		   framwork.Logf("Static node port allocator was not able to reserver nodeport: %d", nodePort)
		   }
```